src/lib/rust.ts

```diff
@@ -120,23 +120,44 @@ const ensureRust = async (): Promise<void> => {
   }
 };
 
-const isRustInstalled = async (): Promise<boolean> => {
-  try {
-    await execPromise({
-      command: CARGO,
-      args: ["--version"],
-    });
+async function isRustInstalled(): Promise<boolean> {
+  if (await isRustInstalledCheck()) {
+    return true;
+  }
 
-    await execPromise({
-      command: RUSTUP,
-      args: ["--version"],
-    });
+  if (process.env.PATH === undefined || process.env.HOME === undefined) {
+    return false;
+  }
```

Comment:
probably you can try to extract home dir with `os.homedir` ?
Main advantage that it's simpler and works across platforms.

Reply:
ok. used homedir, it's probably a bit simpler
but no real reason to care about cross platform cause I fail anyway if on windows and ask user to install rust by themselves

Reply:
I see. But as far as i know we had some plans to support windows someday, right? If we use crossplatform API when possible it will make the window adoption a bit easier. Even if it is a single function right now.